### PR TITLE
avrdude: update to 7.3

### PR DESCRIPTION
**Pull request by sq8kfh**
#### Description

- update to 7.3 version
- update homepage url
- switch to github PortGroup

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 14.2.1 23C71 arm64
Xcode 15.2 15C500b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: cross/avrdude/Portfile [hunk 1]
```diff
@@ -1,62 +1,38 @@
...
+description         AVR Downloader Uploader
+long_description    AVRDUDE - AVR Downloader Uploader - is a program for downloading and uploading \
+                    the on-chip memories of Microchip’s AVR microcontrollers. It can program the Flash \
+                    and EEPROM, and where supported by the programming protocol, it can program fuse \
+                    and lock bits. AVRDUDE also supplies a direct instruction mode allowing one to issue \
+                    any programming instruction to the AVR chip regardless of whether AVRDUDE implements \
+                    that specific feature of a particular chip.
 
-homepage            https://www.nongnu.org/avrdude/
+homepage            https://github.com/avrdudes/avrdude
```
Comment (reneeotten) [inline cross/avrdude/Portfile:20]: this is set already by the "github" PortGroup so can be removed

File: cross/avrdude/Portfile [hunk 2]
```diff
@@ -1,62 +1,38 @@
...
+use_configure       no
 
-configure.args      --disable-parport
+build.cmd           ./build.sh
 
-# https://trac.macports.org/ticket/66722
-compiler.c_standard 2011
+destroot.cmd        cmake
+destroot.pre_args   --install build_darwin
+destroot.post_args  --prefix ${destroot}/${prefix}
```
Comment (reneeotten) [inline cross/avrdude/Portfile:37]: can this use the cmake PortGroup and the default build/destroot commands set by it?

File: cross/avrdude/Portfile [hunk 3]
```diff
@@ -1,62 +1,38 @@
...
+                    that specific feature of a particular chip.
 
-homepage            https://www.nongnu.org/avrdude/
+homepage            https://github.com/avrdudes/avrdude
 
-master_sites        https://download.savannah.gnu.org/releases/avrdude/
-checksums           rmd160  b00adf7211c10cfa1e75f6969cb22e8b863e8671 \
-                    sha256  c0ef65d98d6040ca0b4f2b700d51463c2a1f94665441f39d15d97442dbb79b54 \
-                    size    943631
+fetch.type          git
```
Comment (reneeotten) [inline cross/avrdude/Portfile:22]: this is not correct. Pleade set "github.tarball_from archives" and add the "checksums"

File: cross/avrdude/Portfile (hunk 1 above)
Comment (sq8kfh) [reply cross/avrdude/Portfile]: Fixed

File: cross/avrdude/Portfile (hunk 3 above)
Comment (sq8kfh) [reply cross/avrdude/Portfile]: Fixed

File: cross/avrdude/Portfile (hunk 2 above)
Comment (sq8kfh) [reply cross/avrdude/Portfile:39]: Avrdude has its own build.sh script. To use cmake PortGroup and remove destroot.*, I have to stop using build.sh.

Comment (reneeotten) [reply cross/avrdude/Portfile:39]: Yes, please don't use that if it's not needed. It appears to me it just some convenience script to set variables based on whether it's building on Linux or macOS (and there is differentiates between MacPorts and Homebrew). 

Presumably everything that needs to be set as variables and/or flags is done already by MacPorts when using the `cmake` PortGroup. So please try removing `use_configure no`,  `build.cmd` and all `destroot` stuff. Instead add `PortGroup           cmake 1.1`, remove the explicit `cmake` build dependency and let MacPorts take care of it. It that somehow doesn't work we can look into a bit closer to see what needs to be resolved.

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Comment by pmetzger** [comment]
@sq8kfh Have you addressed all of @reneeotten's comments?

**Comment by sq8kfh** [comment]
Yes

**Pull request merged**